dimagi/commcare-hq
# Multi select web apps UI-- checkboxes

## Product Description

Ticket: [USH-1742](https://dimagi-dev.atlassian.net/browse/USH-1742)
This adds a column of checkboxes (and a "select all" checkbox), and modify the button to say "continue" if it's a multi-select case list

In Progress Screenshots:
<img width="729" alt="image" src="https://user-images.githubusercontent.com/36681924/160668037-9893edee-6754-40df-8e93-6ed008904eac.png">
<img width="628" alt="image" src="https://user-images.githubusercontent.com/36681924/160668099-2af837e1-24ae-4a74-8abd-2f195a2efd0e.png">


## Technical Summary

For the Draft the items I still have remaining:

- [X] when viewing the case list details modify to say "select" and "cancel" rather than continue
- [X] investigate color-changing row when row is selected (investigated felt like I was close to this change for a bit, but realized it was taking too much time-- I'll either leave this to the end of this PR or push to another PR since it's not blocking for the functionality)
- [X] accessibility fixes
- [ ]Change default to false for the multiseelct setting until the logic is fixed in a separate PR

leaving for next continuation of followup PRs:
_- [ ] Check if type is multi-select case list_
_- [ ] add a "continue" button that states how many items are selected_


## Feature Flag

The new multi-select case list flag

## Safety Assurance

### Safety story

This will be behind a flag and will go through QA

### Automated test coverage


No

### QA Plan


I will be requesting QA- note to request accessibility testing on this as well. All checkboxes should be able to be selected via keyboard in addition to the new buttons. The screen reader should specify the label and/or that it's a checkbox as well. Note: to enable checkbox with keyboard, toggle to checkbox with tab key and to select/deselect press the spacebar

### Rollback instructions




### Labels & Review
